### PR TITLE
CAKE-3137 add the scaffolding for class and component testing

### PR DESCRIPTION
**Pull request by drsnyder**
https://wikia-inc.atlassian.net/browse/CAKE-3137

Adds Karma + Mocha + Chai for testing. 

In the first pass at this, I tried using Enzyme for the rendering test but ran into some issues with the helper libraries provided by the community. Either they wouldn't run due to internal errors or `wrapper.find(<classname>)` would only work at the top-most level.

I decided to backtrack and use the simplest approach which is to render the component directly. This may have limitations of it's own down the road if we need to invoke lifecycle methods. We can revisit if necessary.

I added two CLI options for testing:
 - `yarn test` runs a single pass
 - `yarn test:watch` rebuilds, and re-runs the tests

The testing is nowhere near complete. I wanted to go ahead and get this merged so we can start backfilling the tests and add tests as we go.

@Wikia/cake

### CODE REVIEW

File: karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+              rules: [
+                  {
+                      test: /\.js$/,
+                      include: path.resolve(__dirname, './src'),
+                      use: [{
+                          loader: 'babel-loader',
+                          options: {
+                              presets: ['env', 'preact'],
+                              plugins: [
+                                  'transform-class-properties',
```
Comment (JoshuaRogan) [inline karma.conf.js]: Can this use the `.babelrc` file instead?

Comment (drsnyder) [reply karma.conf.js]: Ah. Yes a6bae4f.

File: karma.conf.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+                  },
+                  {
+                      test: /\.s?css$/,
+                      use: [
+                          {
+                              loader: 'style-loader',
+                              options: {
+                                  sourceMap: true,
+                              }
+                          },
```
Comment (drsnyder) [inline karma.conf.js]: This block addressed the issue I was having with the `style` object in `App` and `App_test`.

**Comment by drsnyder** [comment]
I looked at consolidating some of the babel settings. I agree it's lame to copy them but it's also hard to follow when part of the config is included (as we're doing in the FC). For my part, I find it easier to look at them as a contiguous block.

**Comment by drsnyder** [comment]
Added code coverage in 64a967d.

**Comment by drsnyder** [comment]
I changed the `mode` to `'development'`. I added a `debugger` statement to `render` and the code was transpiled but readable.
